oak_functions/loader/src/main.rs

```diff
@@ -107,7 +108,8 @@ impl LookupData {
     /// at a future time.
     async fn refresh(&self) -> anyhow::Result<()> {
         // TODO(#1930): Avoid loading the entire file in memory for parsing.
-        let client = hyper::Client::new();
+        let https = HttpsConnector::new();
```

Comment:
In the rest of the code we use `hyper_rustls::HttpsConnector`, which allows using custom CA store:
https://github.com/project-oak/oak/blob/57d965254fdf4d22d87b7bf0c2863caa109cd851/oak_runtime/src/node/http/client.rs#L309-L316

Reply:
cc @conradgrobler

Reply:
Do we need a custom CA store? @tiziano88 @rbehjati

Reply:
I don't know if @conradgrobler has an opinion, having implemented the CA store for the oak_loader, anyways I don't think for the functions loader it matters that much, since we are not making promises about the confidentiality of the data in terms of TLS certs, we only use it for fetching the lookup data, so I think your version is fine.

Reply:
That said, it may be nice to use hyper_rustls for consistency with the existing code, even if we would not manually add a CA store. WDYT?

Reply:
We needed support for custom CA certificates to run the other examples because we used certificates that we signed ourselves. For this download I assume the production data will usually be hosted somewhere with a proper certificate, so this should not be required here. For test cases it can just use HTTP.

Reply:
AFAICT

- hyper-rustls uses https://github.com/ctz/rustls
- hyper-tls uses https://github.com/sfackler/rust-native-tls

I don't know if we have an opinion already on which to prefer, but presumably we should stick to one everywhere if possible.

Perhaps we should talk to the security team about this too, @anghelcovici could you get in touch with them?

Reply:
Also see https://github.com/project-oak/oak/issues/1254 (it seems the conclusion was to use rustls)

Reply:
Ok, I've switched to hyper_rustls for consistency. Using native roots for now, we can change this if we need.